java-frontend/src/test/java/org/sonar/java/model/JParserSemanticTest.java

```diff
@@ -379,6 +390,18 @@ void declaration_method() {
     assertThat(cu.sema.declarations.get(m.methodBinding)).isSameAs(m);
   }
 
+  @Test
+  void declaration_method_parameterized() {
+    JavaTree.CompilationUnitTreeImpl cu = test("class C { <T> void m(T t) { m(\"hello\"); } }");
+    ClassTreeImpl c = (ClassTreeImpl) cu.types().get(0);
+    MethodTreeImpl m = (MethodTreeImpl) c.members().get(0);
+    MethodInvocationTreeImpl m1Invoc = (MethodInvocationTreeImpl) ((ExpressionStatementTree) m.block().body().get(0)).expression();
+
+    assertThat(m.methodBinding).isNotNull();
+    assertThat(cu.sema.declarations.get(m.methodBinding)).isEqualTo(m);
+    assertThat(cu.sema.usages.get(m.methodBinding)).containsOnly((IdentifierTreeImpl) m1Invoc.methodSelect());
```

Comment:
Similarly to use modifications in JSymbol one need to call `declaration` or `usages` on `cu.sema.methodSymbol((IMethodBinding) identifier.binding)`, which corresponds to `identifier.symbol()` in old implementation. And I'm wondering about existence of such calls.